src/collector.ts

```diff
@@ -142,8 +171,8 @@ class NaiveGomodParser {
 class GomodDependencyCollector implements IDependencyCollector {
     constructor(public classes: Array<string> = ["dependencies"]) {}
 
-    async collect(contents: string): Promise<Array<IDependency>> {
-        let parser = new NaiveGomodParser(contents);
+    async collect(contents: string, manifestFile: string): Promise<Array<IDependency>> {
```

Comment:
manifest file path(uri) can be obtained and passed while instantiating GoModeParser class through the constructor. It would have eliminated changes to rest of the ecosystem interfaces and classes.